gui/tasks/scripts.js

```diff
@@ -0,0 +1,32 @@
+const { parallel, series, src, dest } = require('gulp');
+const envify = require('gulp-envify');
+const ts = require('gulp-typescript');
+
+const TscWatchClient = require('tsc-watch/client');
+
+function makeWatchCompiler(onFirstSuccess) {
+  const compileScripts = function() {
+    const watch = new TscWatchClient();
+    watch.on('first_success', onFirstSuccess);
+    watch.start('--noClear', '--sourceMap', '--project', '.');
+    return watch.tsc;
+  };
+  compileScripts.displayName = 'compile-scripts-watch';
+
+  return compileScripts;
+}
+
+function compileScripts() {
+  const tsProject = ts.createProject('tsconfig.json');
```

Comment:
FYI Too bad `gulp-typescript` does not support incremental compilation for TS, so I had to stick with `TscWatchClient` for development builds, however we still can use `gulp-typescript` for making release builds.